orca-remote-stage/src/main/kotlin/com/netflix/spinnaker/orca/remote/stage/messages/RemoteStageStatusMessageHandler.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.orca.remote.stage.messages
+
+import com.netflix.spinnaker.orca.api.pipeline.models.ExecutionType
+import com.netflix.spinnaker.orca.api.pipeline.remote.messages.RemoteStageMessage
+import com.netflix.spinnaker.orca.api.pipeline.remote.messages.RemoteStageStatusMessage
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import com.netflix.spinnaker.orca.remote.stage.StageExecutionNotFound
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class RemoteStageStatusMessageHandler(
+  val executionRepository: ExecutionRepository
+) : RemoteStageMessageHandler<RemoteStageStatusMessage> {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun supports(message: Class<RemoteStageMessage>): Boolean {
+    return message == RemoteStageStatusMessage::class.java
+  }
+
+  override fun handle(message: RemoteStageStatusMessage) {
+    log.info("Remote stage execution '{}' status received with '{}'", message.stageExecutionId, message.status.toString())
+    val pipelineExecution = executionRepository.retrieve(ExecutionType.PIPELINE, message.pipelineExecutionId)
+    val stageExecution = pipelineExecution.stages.find { it.id == message.stageExecutionId }
+
+    if (stageExecution != null) {
+      stageExecution.status = message.status
+      stageExecution.outputs = message.outputs
+      stageExecution.context["remoteTasks"] = message.tasks
```

Comment:
I would've expected that tasks received from a message would be applied directly to the stage. If left like this, then users would have to open the execution source from the UI to see what was actually going on.

Reply:
Applied to the `stageExecution.tasks`?  I think that runs the risk of queuing the task, and in this case we don't want to do that.  This is more similar to a kato task than an internal Orca task.  I was thinking if we standardize around a key in the context (`remote.tasks`/`remoteTasks` for example) then we can do some additional UI work in Deck to display these tasks to the end user.

Reply:
I see this as just arbitrary context that could be shown back to a user ... `tasks` seems very specific and overloaded. 

I think there's value in having some sort of a contract around what this is (ie. is it a `List<String>` or something else?)

Reply:
I don't think `List<String>` will be good enough - these remote stages are likely going to want to have custom deck UI as well, and will need to pass information that the UI can use to render things as needed.